components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -3671,4 +3672,30 @@ public static boolean isAPIBasedAuthenticationFlow(HttpServletRequest request) {
 
         return Boolean.TRUE.equals(request.getAttribute(FrameworkConstants.IS_API_BASED_AUTH_FLOW));
     }
+
+    /**
+     * Create a shallow copy of the input Identity Provider.
+     *
+     * @param idP Identity Provider.
+     * @return Clone of IDP.
+     */
+    public static IdentityProvider createIdPClone(IdentityProvider idP) {
```

Comment:
Can we use deep clone rather than using Gson. Because doing JSON processing will add extra latency to the flow

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/commit/87aac03e1fe4a099ff0d87cfced2667a1a3f74ad